infra/cardano-sl-infra.cabal

```diff
@@ -212,15 +212,41 @@ test-suite test
   type:             exitcode-stdio-1.0
 
   other-modules:    Spec
+                    Test.Pos.Infra.Arbitrary
+                    Test.Pos.Infra.Arbitrary.Communication
+                    Test.Pos.Infra.Arbitrary.Slotting
+                    Test.Pos.Infra.Arbitrary.Ssc
+                    Test.Pos.Infra.Arbitrary.Txp
+                    Test.Pos.Infra.Arbitrary.Update
+                    Test.Pos.Infra.Bi
                     Test.Pos.Infra.Diffusion.Subscription.StatusSpec
                     Test.Pos.Infra.Diffusion.Subscription.SubscriptionSpec
+                    Test.Pos.Infra.Gen
 
-  build-depends:    async
+  build-depends:    QuickCheck
+                  , async
                   , base
+                  , bytestring
+                  , cardano-sl-binary
+                  , cardano-sl-binary-test
+                  , cardano-sl-core
+                  , cardano-sl-core-test
+                  , cardano-sl-crypto
                   , cardano-sl-crypto-test
                   , cardano-sl-infra
+                  , cardano-sl-ssc
+                  , cardano-sl-ssc-test
+                  , cardano-sl-update-test
+                  , cardano-sl-util-test
                   , containers
-                  , QuickCheck
+                  , generic-arbitrary
+                  , hedgehog
                   , hspec
+                  , kademlia
+                  , universum
 
-  ghc-options:      -threaded
+  ghc-options:         -Wall
+                       -O2
+                       -threaded
+
+  default-extensions:  NoImplicitPrelude
```

Comment:
We really shouldn't be adding `default-extensions`.

Reply:
I think having `NoImplicitPrelude` is fine, since we seem to use `Universum` everywhere.